test/TestImportIncludeRole.py

```diff
@@ -55,56 +57,68 @@
 '''
 
 
-class TestImportIncludeRole(unittest.TestCase):
-    def setUp(self):
-        rulesdir = os.path.join('lib', 'ansiblelint', 'rules')
-        self.rules = RulesCollection([rulesdir])
-
-        # make dir and write role tasks to import or include
-        self.play_root = tempfile.mkdtemp()
-        role_path = os.path.join(self.play_root, 'test-role', 'tasks')
-        os.makedirs(role_path)
-        with open(os.path.join(role_path, 'main.yml'), 'w') as f_main:
-            f_main.write(ROLE_TASKS_MAIN)
-        with open(os.path.join(role_path, 'world.yml'), 'w') as f_world:
-            f_world.write(ROLE_TASKS_WORLD)
-
-    def tearDown(self):
-        shutil.rmtree(self.play_root)
-
-    def _get_play_file(self, playbook_text):
-        with open(os.path.join(self.play_root, 'playbook.yml'), 'w') as f_play:
-            f_play.write(playbook_text)
-        return f_play
-
-    def test_import_role(self):
-        fh = self._get_play_file(PLAY_IMPORT_ROLE)
-        runner = Runner(self.rules, fh.name, [], [], [])
-        results = runner.run()
-        assert 'only when shell functionality is required' in str(results)
-
-    def test_import_role_inline_args(self):
-        fh = self._get_play_file(PLAY_IMPORT_ROLE_INLINE)
-        runner = Runner(self.rules, fh.name, [], [], [])
-        results = runner.run()
-        assert 'only when shell functionality is required' in str(results)
-
-    def test_include_role(self):
-        fh = self._get_play_file(PLAY_INCLUDE_ROLE)
-        runner = Runner(self.rules, fh.name, [], [], [])
-        results = runner.run()
-        assert 'only when shell functionality is required' in str(results)
-        assert 'All tasks should be named' in str(results)
-
-    def test_include_role_inline_args(self):
-        fh = self._get_play_file(PLAY_INCLUDE_ROLE_INLINE)
-        runner = Runner(self.rules, fh.name, [], [], [])
-        results = runner.run()
-        assert 'only when shell functionality is required' in str(results)
-        assert 'All tasks should be named' in str(results)
-
-    def test_invalid_import_role(self):
-        fh = self._get_play_file(PLAY_IMPORT_ROLE_INCOMPLETE)
-        runner = Runner(self.rules, fh.name, [], [], [])
-        with pytest.raises(RuntimeError, match="Failed to find required 'name' key in import_role"):
-            runner.run()
+def _setup_play(tmpdir, playbook_text):
```

Comment:
This should be a fixture

Reply:
Not sure, I use it as a helper to define fixtures (see a few lines below) but I needs to keep its arguments free. Hand you cannot call a fixture function directly, so making this a fixture would defeat the purpose. Though to avoid confusion I renamed the `tmpdir` parameter to `base_dir`.

Reply:
Well, you can call fixture functions by using smth like `request.getfixture('fixturename')` but I guess you cannot pass params explicitly.